src/apint/utils.rs

```diff
@@ -321,33 +366,48 @@ impl ApInt {
     /// - One (`1`) is also called the multiplicative neutral element.
     /// - This operation is more efficient than comparing two instances
     ///   of `ApInt` for the same reason.
+    #[inline]    
     pub fn is_one(&self) -> bool {
         match self.access_data() {
             DataAccess::Inl(digit) => digit == Digit::one(),
             DataAccess::Ext(digits) => {
-                let (last, rest) = digits.split_last()
-                    .expect("An `ApInt` always has at least one digit so calling \
-                             `split_last` on a slice of its digits will never \
-                             return `None`.");
+                let (last, rest) = digits.split_last().unwrap_or_else(|| unreachable!());
```

Comment:
Maybe we should add a safe API for `split_last` since it cannot fail for all digits as one of their invariants is to never be empty.

Reply:
Should I make an issue for that? There are a lot of other `.expect`s around the library that might be eliminated this way.

Reply:
I think it doesn't matter if it also is part of this giantic PR. So just add it if you want but please with proper doc etc. ;)